coderd/workspaceagents.go

```diff
@@ -1653,6 +1653,13 @@ func (api *API) workspaceAgentReportStats(rw http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	if workspace.Deleted {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Workspace has been deleted.",
+		})
+		return
+	}
```

Comment:
It probably makes more sense to do this in GetWorkspaceAgentAndOwnerByAuthToken instead.

Reply:
I am actually surprised we do not already block this. I thought we did invalidate the agent token somewhere :thinking:

Reply:
I moved it into the query, so now it happens in middleware automatically.